src/ai/backend/manager/scheduler/predicates.py

```diff
@@ -128,7 +130,7 @@ async def check_dependencies(
     rows = result.fetchall()
     pending_dependencies = []
     for row in rows:
-        if row.result != SessionResult.SUCCESS:
+        if row.result != SessionResult.SUCCESS or row.status != SessionStatus.TERMINATED:
```

Comment:
Is it okay not to cover `CANCELLED` session status here?

Reply:
I think those which depend on `CANCELLED` sessions are not supposed to be started.

Reply:
They rather be notified and `CANCELLED` either?

Reply:
Let's keep this as is for now and create a patch if same issue occurs for `CANCELLED` sessions.